From 2f819641f6e973ac22a8c9aea3b4b50daeb771d9 Mon Sep 17 00:00:00 2001
From: Yaniv Levinsky <ylevinsky@habana.ai>
Date: Sun, 18 Dec 2022 16:40:27 +0200
Subject: [PATCH] [SIV-1515] gaudi2: move mgmt reloc section to hbm

Move the reloc section of the ARC1 MGMT image to hbm.

This is a part of the SIV ARC1 proof of concept for running from HBM.

Change-Id: I3c116a754c3f7091913fa55ef7c27ebf62e28ae8
Signed-off-by: Yaniv Levinsky <ylevinsky@habana.ai>
---

diff --git a/board/habanalabs/common/uboot-zephyr_reloc.its b/board/habanalabs/common/uboot-zephyr_reloc.its
index 17c9aef..879908b 100644
--- a/board/habanalabs/common/uboot-zephyr_reloc.its
+++ b/board/habanalabs/common/uboot-zephyr_reloc.its
@@ -108,7 +108,7 @@
 			description = "Zephyr mgmt RELOC";
 			data = /incbin/("@HL_DEV@-zephyr-mgmt-reloc.bin.le");
 			type = "standalone";
-			arch = "arc";
+			arch = "hbm";
 			compression = "none";
 			loadaddr = <@ARC_MGMT_RELOC_HI@ @ARC_MGMT_RELOC_LO@>;
 			hash {
diff --git a/board/habanalabs/gaudi2/post-image-fit.sh b/board/habanalabs/gaudi2/post-image-fit.sh
index c83d313..16f8549 100644
--- a/board/habanalabs/gaudi2/post-image-fit.sh
+++ b/board/habanalabs/gaudi2/post-image-fit.sh
@@ -20,8 +20,8 @@
 ARC_MGMT_DCCM_LO=0xF7060000
 ARC_MGMT_ICCM_HI=0x1000007F
 ARC_MGMT_ICCM_LO=0xF7040000
-ARC_MGMT_RELOC_HI=0x1000007F
-ARC_MGMT_RELOC_LO=0xF7012400
+ARC_MGMT_RELOC_HI=0x10010000
+ARC_MGMT_RELOC_LO=0x00000000
 ARC_PID_DCCM_HI=0x1000007F
 # Should be aligned with arcpid DCCM_START
 ARC_PID_DCCM_LO=0xF7020000
diff --git a/board/habanalabs/gaudi2_arc/asic_mgmt.bins b/board/habanalabs/gaudi2_arc/asic_mgmt.bins
index 6de651b..96cb770 100644
--- a/board/habanalabs/gaudi2_arc/asic_mgmt.bins
+++ b/board/habanalabs/gaudi2_arc/asic_mgmt.bins
@@ -2,4 +2,4 @@
 iccm,0x27040000,0x80,16
 dccm,0x80000000,0x20,16
 sram,0x27fe4800,0x2e,4
-reloc,0x27012400,0x17,4
+reloc,0x40000000,0x27,4
